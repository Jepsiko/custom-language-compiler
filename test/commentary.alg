
CO

This is a multiple line comment

CO

CO
    co this is a nested short comment
CO

CO
CO
This is not a nested long comment
CO
CO

CO abriCO

begin
    read(a);
    read(b)
end

CO

begin
    read(c);
    read(d)
end