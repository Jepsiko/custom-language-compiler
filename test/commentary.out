token: This	lexical unit: VARNAME
token: is	lexical unit: VARNAME
token: a	lexical unit: VARNAME
token: not	lexical unit: NOT
token: nested	lexical unit: VARNAME
token: long	lexical unit: VARNAME
token: comment	lexical unit: VARNAME
token: begin	lexical unit: BEG
token: read	lexical unit: READ
token: (	lexical unit: LEFT_PARENTHESIS
token: c	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: ;	lexical unit: SEMICOLON
token: read	lexical unit: READ
token: (	lexical unit: LEFT_PARENTHESIS
token: d	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: end	lexical unit: END

Identifiers
This	14
a	14
c	28
comment 14
d   29
is  14
long    14
nested  14
